src/textEditor.ts

```diff
@@ -224,3 +224,61 @@ export class TextEditor {
   }
 }
 
+/**
+ * Directions in the view for editor scroll command.
+ */
+export const EditorScrollDirection = {
+  Up: 'up',
+  Down: 'down',
+};
+
+/**
+ * Units for editor scroll 'by' argument
+ */
+export const EditorScrollByUnit = {
+  Line: 'line',
+  WrappedLine: 'wrappedLine',
+  Page: 'page',
+  HalfPage: 'halfPage'
+};
+
+/**
+ * Values for reveal line 'at' argument
+ */
+export const RevealLineAtArgument = {
+  Top: 'top',
+  Center: 'center',
+  Bottom: 'bottom'
+};
+
+/**
+ * Positions in the view for cursor move command.
+ */
+export const CursorMovePosition = {
```

Comment:
yes, string union type can easily validate string value assignment but if we use that, then we still need to write plain text string while initialization in `action.ts`, which is what we want to avoid, right?